db/migrate/20240208170628_add_external_id_to_group.rb

```diff
@@ -0,0 +1,12 @@
+class AddExternalIdToGroup < ActiveRecord::Migration[7.1]
+  def change
+    # Rubcop will complain that we are making a column not null without a default value
+    # which wouldn't work if we already had data in the table. We are going to ignore this
+    # because we don't have any data in the table yet.
+
+    # rubocop:disable Rails/NotNullColumn
```

Comment:
Rubocop is slightly over protective here so we disable it. 

Rubocop is worrying about the consequences of adding a not null constraint to an existing column where not all rows have a value. This would be awkward because the migration would fail.

We don't have to worry because this is a new table even though the migrations are split here. There won't be any groups without this value and we don't ever want it not to be set. There is no obvious default value as we have taken the decision to generate the external_id in ruby.